subsys/net/lib/websocket/websocket.c

```diff
@@ -862,11 +862,71 @@ static int websocket_parse(struct websocket_context *ctx, struct websocket_buffe
 	return parsed_count;
 }
 
+#if !defined(CONFIG_NET_TEST)
+static int wait_rx(int sock, int timeout)
+{
+	struct zsock_pollfd fds = {
+		.fd = sock,
+		.events = ZSOCK_POLLIN,
+	};
+	int ret;
+
+	ret = zsock_poll(&fds, 1, timeout);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (ret == 0) {
+		/* Timeout */
+		return -EAGAIN;
+	}
+
+	if (ret == 1) {
```

Comment:
Do we need to check this, I mean we are polling one specific fd so the return value at this point must be 1

Reply:
True, it's redundant here, I've removed the check. Thanks!